src/ndarray/ndarray_function.cc

```diff
@@ -142,7 +142,10 @@ void GetUniqueRspRowIdx(const std::vector<NDArray>& nds,
   uniq_row_idx->resize(it - uniq_row_idx->begin());
 }
 
-void ElementwiseSumRsp(mshadow::Stream<cpu>* s, const std::vector<NDArray>& nds, NDArray* out) {
+void ElementwiseSumRsp(mshadow::Stream<cpu>* s,
+                       Resource rsc,
```

Comment:
`get_space_typed` has a `const` qualifier for the function body, so passing by `const Resource&` should also compile. If that's the case, please pass `rsc` by const reference here.